### DISCUSSION TITLE
Callback for data change

### DISCUSSION THREAD

**Discussion by ghost** [pmndrs/valtio#119] [Q&A] [answered]
Is there a way to create a callback for data change?
For example I have a proxy const state = proxy({ count: 0, text: 'hello' })

but I want to have a function somethingChanged to call when state will get updated.
Can I do this?

**Comment by ghost** [comment, marked as answer]
[SOLVED] useEffect + useSnapshot works perfectly for a react app

**Comment by dai-shi** [comment]
You may want to try `subscribe` (or `subscribeKey` in utils).
